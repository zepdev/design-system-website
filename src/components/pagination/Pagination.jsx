import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import IconButton from '../button/IconButton'
import PreviousPageIcon from '../icons/PreviousPageIcon'
import NextPageIcon from '../icons/NextPageIcon'
import ChevronLeftIcon from '../icons/ChevronLeftIcon'
import ChevronRightIcon from '../icons/ChevronRightIcon'

const Pagination = ({ className: classNameProp, count }) => {
  const numbers = new Array(4).fill(0)
  console.log(numbers)
  return (
    <div className={classnames('zep-pagination', classNameProp)}>
      <IconButton>
        <PreviousPageIcon className="zep-button__icon" />
      </IconButton>
      <IconButton>
        <ChevronLeftIcon className="zep-button__icon" />
      </IconButton>
      {numbers.map((elem, idx) => (
        <button key={`pagination${ idx }`} className="zep-button zep-button-pagination">
          {idx + 1}
        </button>
      ))}
      <IconButton>
        <ChevronRightIcon className="zep-button__icon" />
      </IconButton>
      <IconButton>
        <NextPageIcon className="zep-button__icon" />
      </IconButton>
    </div>
  )
}

Pagination.propTypes = {
  className: PropTypes.object,
  children: PropTypes.array,
}

export default Pagination
