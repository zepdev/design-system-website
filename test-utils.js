import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
