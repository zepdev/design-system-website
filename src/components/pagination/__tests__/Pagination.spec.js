import React from "react"
import { render } from "../../../../test-utils"
import Pagination from "../Pagination"

describe("Pagination", () => {
  it("renders correctly", () => {
    const mockProps = {
      onPageChange: jest.fn(),
      currentPage: 1,
      pagesToDisplay: 5,
      pages: 5,
    }
    const component = render(<Pagination {...mockProps} />)
    expect(component).toMatchSnapshot()
  })
})
