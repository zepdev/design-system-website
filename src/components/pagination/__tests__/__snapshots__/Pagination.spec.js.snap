// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="zep-pagination"
      >
        <button
          class="zep-button zep-button--icon"
          data-testid="paginationTo0"
          disabled=""
        >
          <svg
            aria-labelledby="title_menu"
            class="zep-button__icon zep-button__icon"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="title_menu"
            >
              PreviousPage
            </title>
            <path
              d="M23.213 22.107l-6.107-6.107 6.107-6.12-1.88-1.88-8 8 8 8 1.88-1.893zM10.667 8h-2.667v16h2.667v-16z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="zep-button zep-button--icon"
          data-testid="paginationBack"
          disabled=""
        >
          <svg
            aria-labelledby="title_menu"
            class="zep-button__icon zep-button__icon"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="title_menu"
            >
              ChevronLeft
            </title>
            <path
              d="M20.547 22.107l-6.107-6.107 6.107-6.12-1.88-1.88-8 8 8 8z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="zep-button zep-button-pagination zep-pagination--selected"
          data-testid="paginationButton1"
        >
          1
        </button>
        <button
          class="zep-button zep-button-pagination"
          data-testid="paginationButton2"
        >
          2
        </button>
        <button
          class="zep-button zep-button-pagination"
          data-testid="paginationButton3"
        >
          3
        </button>
        <button
          class="zep-button zep-button-pagination"
          data-testid="paginationButton4"
        >
          4
        </button>
        <button
          class="zep-button zep-button-pagination"
          data-testid="paginationButton5"
        >
          5
        </button>
        <button
          class="zep-button zep-button--icon"
          data-testid="paginationForward"
        >
          <svg
            aria-labelledby="title_menu"
            class="zep-button__icon zep-button__icon"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="title_menu"
            >
              ChevronRight
            </title>
            <path
              d="M11.453 22.107l6.107-6.107-6.107-6.12 1.88-1.88 8 8-8 8z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="zep-button zep-button--icon"
          data-testid="paginationToLast"
        >
          <svg
            aria-labelledby="title_menu"
            class="zep-button__icon zep-button__icon"
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="title_menu"
            >
              NextPage
            </title>
            <path
              d="M8 22.107l6.107-6.107-6.107-6.12 1.88-1.88 8 8-8 8-1.88-1.893zM20.547 8h2.667v16h-2.667v-16z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="zep-pagination"
    >
      <button
        class="zep-button zep-button--icon"
        data-testid="paginationTo0"
        disabled=""
      >
        <svg
          aria-labelledby="title_menu"
          class="zep-button__icon zep-button__icon"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title_menu"
          >
            PreviousPage
          </title>
          <path
            d="M23.213 22.107l-6.107-6.107 6.107-6.12-1.88-1.88-8 8 8 8 1.88-1.893zM10.667 8h-2.667v16h2.667v-16z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="zep-button zep-button--icon"
        data-testid="paginationBack"
        disabled=""
      >
        <svg
          aria-labelledby="title_menu"
          class="zep-button__icon zep-button__icon"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title_menu"
          >
            ChevronLeft
          </title>
          <path
            d="M20.547 22.107l-6.107-6.107 6.107-6.12-1.88-1.88-8 8 8 8z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="zep-button zep-button-pagination zep-pagination--selected"
        data-testid="paginationButton1"
      >
        1
      </button>
      <button
        class="zep-button zep-button-pagination"
        data-testid="paginationButton2"
      >
        2
      </button>
      <button
        class="zep-button zep-button-pagination"
        data-testid="paginationButton3"
      >
        3
      </button>
      <button
        class="zep-button zep-button-pagination"
        data-testid="paginationButton4"
      >
        4
      </button>
      <button
        class="zep-button zep-button-pagination"
        data-testid="paginationButton5"
      >
        5
      </button>
      <button
        class="zep-button zep-button--icon"
        data-testid="paginationForward"
      >
        <svg
          aria-labelledby="title_menu"
          class="zep-button__icon zep-button__icon"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title_menu"
          >
            ChevronRight
          </title>
          <path
            d="M11.453 22.107l6.107-6.107-6.107-6.12 1.88-1.88 8 8-8 8z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="zep-button zep-button--icon"
        data-testid="paginationToLast"
      >
        <svg
          aria-labelledby="title_menu"
          class="zep-button__icon zep-button__icon"
          height="32"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title_menu"
          >
            NextPage
          </title>
          <path
            d="M8 22.107l6.107-6.107-6.107-6.12 1.88-1.88 8 8-8 8-1.88-1.893zM20.547 8h2.667v16h-2.667v-16z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
